### DISCUSSION TITLE
Heterogeneous permeability data file

### DISCUSSION THREAD

**Discussion by Joseph-0123** [idaholab/moose#22164] [Q&A General] [answered]
Dear MOOSE developers and users,

I mesh a vertical 3D block (x:200m;y:5km; z: 5km ) by tetrahedrons using Gmsh. So I don't have regular structured grids.

I want to create a heterogeneous permeability data file for my model. I learn the [Heterogeneous models](https://mooseframework.inl.gov/modules/porous_flow/heterogeneous_models.html) . It is helpful for me. But I also need the .data file for my model.

Could you please tell me which open source software/tool can creat such .data file?

Thank you in advance.

Cheer,

Joseph

**Comment by WilkAndy** [comment, marked as answer]
The fundamentals of this file usually come from other modelling, eg, statistical modelling of the subsurface, that is based on field experiments and/or lab data.  This gives you some knowledge of the statistics of the permeability distribution (eg mean, standard deviation).   Alternatively, the information could come from field measurements that you want to interpolate, for instance, you know the permeability close to some well-screens, and you need to interpolate that data to other places in the model.

In any case, assuming you have some notion of the permeability field that you want to create, you must create the .data file yourself.  I imagine that usually people use python, because it is open-source and pretty universally used (you can publish your software code to generate the permeability field).  Alternatively, there are probably proprietary packages like Leapfrog, Geomodeller, GoCAD, etc, that can output this information.

a

**Reply by Joseph-0123** [reply]
Thanks for your guidance. Andy. I am trying to create this data using Python.

Joseph
